src/main/java/Duke.java

```diff
@@ -19,20 +21,90 @@ public static void main(String[] args) {
             if ("list".equalsIgnoreCase(userInput)) {
                 System.out.println("____________________________________________________________");
                 System.out.println("Here are the tasks in your list:");
-                for (int i = 0; i < tasks.size(); i++) {
-                    Task task = tasks.get(i);
-                    System.out.println((i + 1) + ".[" + task.getStatusIcon() + "] " + task.description);
+                for (int i = 0; i < taskCount; i++) {
+                    Task task = tasks[i];
+                    System.out.println((i + 1) + "." + task);
                 }
                 System.out.println("____________________________________________________________");
             } else if (userInput.startsWith("mark ")) {
-                processTask(userInput, tasks, true);
+                processTask(userInput, tasks, true, taskCount);
             } else if (userInput.startsWith("unmark ")) {
-                processTask(userInput, tasks, false);
-            } else {
+                processTask(userInput, tasks, false, taskCount);
+            }else if (userInput.startsWith("todo ")) {
+                if (taskCount >= 100) {
+                    System.out.println("You have reached the task limit. Cannot add more tasks!");
+                    continue;
+                }
+
+                String description = userInput.substring(5).trim();
+                Todo newTodo = new Todo(description);
+                tasks[taskCount] = newTodo;
+                taskCount++;
+
+                System.out.println("____________________________________________________________");
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + newTodo);
+                System.out.println("Now you have " + taskCount + " tasks in the list.");
+                System.out.println("____________________________________________________________");
+            } else if (userInput.startsWith("deadline ")) {
```

Comment:
Also, adopt 'KISSing', make your code simple. For example, you can isolate out some arguments as isolated class file to avoid long winded function in the main class file.